ambry-replication/src/main/java/com.github.ambry.replication/DiskTokenPersistor.java

```diff
@@ -62,6 +67,10 @@ protected void persist(String mountPath, List<ReplicaTokenInfo> tokenInfoList) t
       logger.debug("Completed writing replica tokens to file {}", actual.getAbsolutePath());
     } catch (IOException e) {
       logger.error("IO error while persisting tokens to disk {}", temp.getAbsoluteFile());
+      // check disk state in storageManager. If there is a hardware issue, persistor should skip this bad disk next time.
+      if (!storageManager.isDiskAvailableAtMountPath(mountPath)) {
+        mountPathsToSkip.add(mountPath);
```

Comment:
Do we need logic to remove the mount path when the disk comes back up, or can we assume the server will be restarted in that case?

Reply:
I guess if a disk is bad, we might have to replace the disk and restart server so that mount path of the replicas get changed. Or we might have to trigger replica update via helix ideal state change.

Reply:
Thanks for both inputs. Right now, when replacing a bad disk, we have to restart the server and `mountPathsToSkip` gets re-initialized. If in the future, hot swap is supported, we can consider dynamically removing the path out of `mountPathsToSkip`.